the trait `std::fmt::Display` is not implemented for `tracing_core::field::DisplayValue<&std::string::String>`

## Bug Report

### Version
0.1.18

### Description

Below is sample code that compiles fine with:
```toml
[dependencies]
tracing = {version = "0.1", default-features = false, features = [
  "attributes",
  "std",
]}
tungstenite = "0.11.0"
```

```rust
use tracing::trace;
use tungstenite::protocol::Message;

fn main() {
  let msg = Message::Text("blah".into());
  trace!(message = display(&msg));
}
```

The moment I enable the "log" feature the build breaks, complaining about some missing `Display`, which is provided.

```patch
--- Cargo.toml
+++ Cargo.toml
@@ -10,5 +10,6 @@ edition = "2018"
 tracing = {version = "0.1", default-features = false, features = [
   "attributes",
   "std",
+  "log",
 ]}
 tungstenite = "0.11.0"
```

```
error[E0277]: `tracing_core::field::DisplayValue<&tungstenite::protocol::message::Message>` doesn't implement `std::fmt::Display`
 --> src/main.rs:6:20
  |
6 |   trace!(message = display(&msg));
  |                    ^^^^^^^^^^^^^ `tracing_core::field::DisplayValue<&tungstenite::protocol::message::Message>` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `tracing_core::field::DisplayValue<&tungstenite::protocol::message::Message>`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required by `std::fmt::Display::fmt`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `foo`.

To learn more, run the command again with --verbose.
```

--- a/tracing-core/src/field.rs
+++ b/tracing-core/src/field.rs
@@ -495,6 +495,12 @@ impl<T: fmt::Display> fmt::Debug for DisplayValue<T> {
     }
 }
 
+impl<T: fmt::Display> fmt::Display for DisplayValue<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
 // ===== impl DebugValue =====
 
 impl<T: fmt::Debug> crate::sealed::Sealed for DebugValue<T> {}

